src/components/AttachmentPicker/index.native.js

```diff
@@ -88,160 +85,60 @@ function getDataForUpload(fileData) {
-            },
-            shortcutConfig.descriptionKey,
-            shortcutConfig.modifiers,
-            true,
-        );
-    }
-
-    removeKeyboardListener() {
-        if (!this.unsubscribeEnterKey) {
-            return;
-        }
-        this.unsubscribeEnterKey();
-    }
-
-    /**
-     * Handles the image/document picker result and
-     * sends the selected attachment to the caller (parent component)
-     *
-     * @param {Array<ImagePickerResponse|DocumentPickerResponse>} attachments
-     * @returns {Promise}
-     */
-    pickAttachment(attachments = []) {
-        if (attachments.length === 0) {
-            return;
-        }
-
-        const fileData = _.first(attachments);
-
-        if (fileData.width === -1 || fileData.height === -1) {
-            this.showImageCorruptionAlert();
-            return;
-        }
-
-        return getDataForUpload(fileData)
-            .then((result) => {
-                this.completeAttachmentSelection(result);
-            })
-            .catch((error) => {
-                this.showGeneralAlert(error.message);
-                throw error;
-            });
-    }
+function AttachmentPicker(props) {
+    const completeAttachmentSelection = useRef();
+    const onModalHide = useRef();
+    const keyboardListener = useRef();
 
     /**
      * Inform the users when they need to grant camera access and guide them to settings
      */
-    showPermissionsAlert() {
+    function showPermissionsAlert() {
         Alert.alert(
-            this.props.translate('attachmentPicker.cameraPermissionRequired'),
-            this.props.translate('attachmentPicker.expensifyDoesntHaveAccessToCamera'),
+            props.translate('attachmentPicker.cameraPermissionRequired'),
+            props.translate('attachmentPicker.expensifyDoesntHaveAccessToCamera'),
             [
                 {
-                    text: this.props.translate('common.cancel'),
+                    text: props.translate('common.cancel'),
                     style: 'cancel',
                 },
                 {
-                    text: this.props.translate('common.settings'),
+                    text: props.translate('common.settings'),
                     onPress: () => Linking.openSettings(),
                 },
             ],
             {cancelable: false},
         );
     }
 
+    /**
+     * A generic handling when we don't know the exact reason for an error
+     *
+     */
+    function showGeneralAlert() {
+        Alert.alert(props.translate('attachmentPicker.attachmentError'), props.translate('attachmentPicker.errorWhileSelectingAttachment'));
+    }
+
     /**
      * Common image picker handling
      *
      * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
      * @returns {Promise<ImagePickerResponse>}
      */
-    showImagePicker(imagePickerFunc) {
+    function showImagePicker(imagePickerFunc) {
```

Comment:
Can we simplify this function? It looks like a callback hell. Moreover many nested logic.

Reply:
Yeah, it is fine to keep it for now.